### DISCUSSION TITLE
Importing pnpm-lock files from individual projects to a monorepo root?

### DISCUSSION THREAD

**Discussion by lflo5727** [pnpm/pnpm#6167] [Q&A]
I have a number of PNPM projects that I want to move into a single monorepo with PNPM. However, since they are already using PNPM, they have their respective pnpm-lock.yaml files. 
As I understand it, `pnpm import` only works to import lock files from other package managers such as Yarn, and not PNPM lock files. 

While I find that PNPM can support each project having its own lock file, it slows it down as each project then has its own virtual store. We have compatibility issues and need to import the lock files for our projects.
Is there anyway to import individual project pnpm-lock files into a the monorepo root lock file?

**Comment by jfirebaugh** [comment]
I think this is essentially the same question as I asked on Stack Overflow: ["How can I add an existing directory to a pnpm workspace while preserving its existing package locks?"](https://stackoverflow.com/questions/77434982/)

**Comment by PandelisZ** [comment]
This would be a useful feature 🙏 we're encountering the need for this too
